kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -151,15 +158,36 @@ public boolean healthCheck() {
     baseStreamsConfig.put(DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
 
     // Default producer configurations
-    baseStreamsConfig.put(producerPrefix(LINGER_MS_CONFIG), "2000");
-    baseStreamsConfig.put(producerPrefix(BATCH_SIZE_CONFIG), "2097152");
+    // Set acks to all for high availability and prevent dataloss
+    // default = 1
+    baseStreamsConfig.put(producerPrefix(ACKS_CONFIG), "all");
+    // Increase linger.ms for better throughput
+    // default = 100
+    baseStreamsConfig.put(producerPrefix(LINGER_MS_CONFIG), "500");
+    // Increase batch.size for better throughput
+    // default = 16384
+    baseStreamsConfig.put(producerPrefix(BATCH_SIZE_CONFIG), "524288");
+    // Enable compression on producer for better throughput
+    // default - none
     baseStreamsConfig.put(producerPrefix(COMPRESSION_TYPE_CONFIG), CompressionType.GZIP.name);
-    baseStreamsConfig.put(producerPrefix(MAX_REQUEST_SIZE_CONFIG), "10485760");
+    // Allows to send larger messages to broker
+    // Requires tuning on broker, topic and consumer application as well
+    // default = 1048576
+    baseStreamsConfig.put(producerPrefix(MAX_REQUEST_SIZE_CONFIG), "1048576");
 
     // Default consumer configurations
+    // Increase to enable receiving large messages
+    // default = 1048576
+    baseStreamsConfig.put(consumerPrefix(MAX_PARTITION_FETCH_BYTES_CONFIG), "1048576");
```

Comment:
Here as well default and set value is same - `1048576`

Reply:
Will remove this one too.